backend/routes/query.py

```
@@ -28,9 +28,25 @@
+        return JSONResponse(content={"rate_limits": status})
+    except Exception as e:
+        print(f"Error getting rate limits: {e}")
+        return JSONResponse(
+            content={"error": "Failed to get rate limit status", "rate_limits": {}},
+            status_code=500
+        )
+
+
 @router.post("/query")
 @limiter.limit(AppConfig.RATE_LIMIT)
 async def query_endpoint(request: Request, query: Query, services: dict = Depends(get_services)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)